File: typescript-analysis/src/services/OrderService.ts
import { Order } from '../entities/Order';
import { OrderRepository } from '../repositories/OrderRepository';

export class OrderService {
  private repository: OrderRepository;

  constructor() {
    this.repository = new OrderRepository();
  }

  async create(entity: Order): Promise<Order> {
    return await this.repository.save(entity);
  }

  async findById(id: string): Promise<Order | null> {
    return await this.repository.findById(id);
  }

  async findAll(): Promise<Order[]> {
    return await this.repository.findAll();
  }

  async update(id: string, entity: Partial<Order>): Promise<Order | null> {
    const existing = await this.repository.findById(id);
    if (!existing) return null;

    Object.assign(existing, entity);
    return await this.repository.save(existing);
  }

  async delete(id: string): Promise<void> {
    await this.repository.delete(id);
  }
}

File: complete-test-extracted/service/OrderService.java
package com.test.complete.service;

import com.test.complete.entity.Order;
import com.test.complete.repository.OrderRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import java.util.List;
import java.util.Optional;

import com.test.complete.enums.OrderStatus;

@Service
@RequiredArgsConstructor
@Transactional
public class OrderService {

    private final OrderRepository repository;

    public List<Order> findAll() {
        return repository.findAll();
    }

    public Optional<Order> findById(Long id) {
        return repository.findById(id);
    }

    public Order save(Order entity) {
        return repository.save(entity);
    }

    public void deleteById(Long id) {
        repository.deleteById(id);
    }

    public Order suspendOrder(Long id) {
        Order entity = repository.findById(id)
            .orElseThrow(() -> new RuntimeException("Order not found with id: " + id));
        entity.suspend();
        return repository.save(entity);
    }

    public Order activateOrder(Long id) {
        Order entity = repository.findById(id)
            .orElseThrow(() -> new RuntimeException("Order not found with id: " + id));
        entity.activate();
        return repository.save(entity);
    }

}
